Source/WebCore/loader/LocalFrameLoaderClient.h

```diff
@@ -217,7 +217,9 @@ class WEBCORE_EXPORT LocalFrameLoaderClient : public FrameLoaderClient {
     virtual void updateGlobalHistory() = 0;
     virtual void updateGlobalHistoryRedirectLinks() = 0;
 
-    virtual bool shouldGoToHistoryItem(HistoryItem&) const = 0;
+    virtual bool shouldGoToHistoryItem(HistoryItem&, bool sameDocumentNavigation) const = 0;
```

Comment:
Ideally, the second parameter should really be an enum class:
```
enum class IsSameDocumentNavigation : bool { No, Yes };
```

as we normally frown upon such boolean parameters, which can make call sites less readable.